### PR TITLE
Removing hash and allowing for IUPAC codes

### PR DESCRIPTION
**Pull request by jdaeth274**
- Removes #s from isolate names
- Allows for full IUPAC dictionary to be parsed for RAxML-NG

### CODE REVIEW

File: python/gubbins/ValidateFastaAlignment.py [hunk 1]
```diff
@@ -40,14 +43,15 @@ def does_each_sequence_have_a_name_and_genomic_data(self):
                   return False
                 if re.search('[^ACGTNacgtn-]', str(record.seq))  != None:
                   print("Error with the input FASTA file: One of the sequences contains odd characters, only ACGTNacgtn- are permitted")
+                  print(record.id)
                   return False
```
Comment (nickjcroucher) [inline python/gubbins/ValidateFastaAlignment.py:47]: Might be worth clarifying this error - just saying something like "record.id + ' contains a non-IUPAC character'" or similar

File: python/gubbins/common.py [hunk 2]
```diff
@@ -296,6 +296,7 @@ def parse_and_run(input_args, program_description=""):
                            sequence_reconstruction_command])
             os.chdir(temp_working_dir)
             try:
+                print()
                 subprocess.check_call(sequence_reconstruction_command, shell=True)
```
Comment (nickjcroucher) [inline python/gubbins/common.py]: Is this `print()` doing anything?

File: python/gubbins/common.py [hunk 3]
```diff
@@ -325,6 +326,13 @@ def parse_and_run(input_args, program_description=""):
...
             shutil.copyfile(base_filename + ".start", gaps_alignment_filename)
             reinsert_gaps_into_fasta_file(joint_sequences_filename, gaps_vcf_filename, gaps_alignment_filename)
+            print("~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~")
+            print("~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~")
+            print(gaps_alignment_filename)
+            print("~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~")
+            print("~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~")
+            # if input_args.seq_recon == "raxmlng":
+            #     remove_internal_nodes_from_alignment(gaps_alignment_filename)
             if not os.path.exists(gaps_alignment_filename) \
```
Comment (nickjcroucher) [inline python/gubbins/common.py]: Probably don't need this debugging statement?

File: python/gubbins/common.py [hunk 4]
```diff
@@ -864,6 +874,18 @@ def transfer_internal_node_labels_to_tree(source_tree_filename, destination_tree
...
+    start_cmd = "grep -n '^>internal_Node1$' " + input_filename + " | sed 's/\:.*$//g' "
+    start_num = subprocess.check_output(start_cmd, shell=True)
+    start_num = str(int(start_num.strip().decode("utf-8")) - 1)
+
+    # Now just use head to take all the lines before this
+    head_cmd = "head -n " + start_num + " " + input_filename + " > tmp_aln.txt && mv tmp_aln.txt " + input_filename
+    try:
+        subprocess.check_call(head_cmd, shell=True)
+    except subprocess.SubprocessError:
+        sys.exit("Command to remove internal nodes from raxmlng ancestral alignment failed")
```
Comment (nickjcroucher) [inline python/gubbins/common.py:870]: Debugging statements?

**Review by nickjcroucher — CHANGES_REQUESTED** [review]
Thanks Josh - I think there's a print statement that can be clarified, and a couple of debugging messages that need removing - looks good other than that!

File: python/gubbins/ValidateFastaAlignment.py (hunk 1 above)
Comment (jdaeth274) [reply python/gubbins/ValidateFastaAlignment.py:47]: Thanks Nick, I've extended that error message

File: python/gubbins/common.py (hunk 2 above)
Comment (jdaeth274) [reply python/gubbins/common.py]: No, removed now

File: python/gubbins/common.py (hunk 3 above)
Comment (jdaeth274) [reply python/gubbins/common.py]: Removed

File: python/gubbins/common.py (hunk 4 above)
Comment (jdaeth274) [reply python/gubbins/common.py:870]: Old function that wasn't needed upon further investigation, removed now

**Review by nickjcroucher — APPROVED** [review]

**Pull request merged**